### ISSUE TITLE
Office UI Fabric People Picker Not Supporting Sharepoint group

### ISSUE THREAD

**Issue by vaishali-udupa** [SharePoint/sp-dev-fx-controls-react#161]
<PeoplePicker  context={this.props.webpartContext}
            titleText="Author's" personSelectionLimit={3}
            groupName={""} // Leave this blank in case you want to filter from all users
            showtooltip={true} 
            selectedItems={this._getPeoplePickerItems}
            showHiddenInUI={false}
            principleTypes={[PrincipalType.User,PrincipalType.SharePointGroup]} /> 

[ ] Question
I am not able to get the Sharepoint groups using the ,PrincipalType.SharePointGroup.

Thanks!
Vaishali

**Comment by vlad-nvs** [comment]
Category
[x ] Bug

I'm also have this issue. Version 1.10.0

After short investigation I can suppose that '/_api/web/siteusers' with filter (PrincipalType eq 8) which used inside '_thisLoadUsers' method in PeoplePickerComponent does not return any SharePoint Group. 

Using additional call with '/_api/web/sitegroups' (in case if SharePoint Group should be retrieved) can fix issue.

Thanks
Vladyslav

**Comment by sympmarc** [comment]
Probably connected: in version 1.11.0 beta, specifying the `groupName` doesn't seem to limit the people retrieved. This may be a regression in the re-architecture in 1.11.0.

Example:
`groupName={"Shipping & Logistics Team Members"}`

**Comment by estruyf** [comment]
Seems to be indeed a limitation in the `siteUsers` API. When using the `_api/SP.UI.ApplicationPages.ClientPeoplePickerWebServiceInterface.clientPeoplePickerSearchUs` API (when not applying the absoluteWebUrl), the `groupName` was indeed ignored.

I just did an update to the library to always use the `_api/SP.UI.ApplicationPages.ClientPeoplePickerWebServiceInterface.clientPeoplePickerSearchUs` API and take the `groupName` into account when provided and will also allow you to search for SharePoint groups. Same logic as the people picker in a list is implemented. A new beta version will be available in a matter of minutes.

**Comment by sympmarc** [comment]
I installed the latest, and I'm still seeing everyone in the picker, even when I include:
```
              groupName={"Shipping & Logistics Team Members"}
              webAbsoluteUrl={"https://tenant.sharepoint.com/sites/ShippingLogisticsTeam"}
```
Did you push your changes to @pnp/spfx-controls-react@next?

**Comment by estruyf** [comment]
Changes are indeed available on the latest. Normally you should now see a call happening to the `_api/SP.UI.ApplicationPages.ClientPeoplePickerWebServiceInterface.clientPeoplePickerSearchUs` API for the group name you provided. If the group is not found, control will be disabled and shows a message. So it could be that you have an old version.

Try by installing it like this: `npm i @pnp/spfx-controls-react@1.11.0-beta.96885fc`

**Comment by sympmarc** [comment]
Yup, working for me, too. The dread "developer error". Than ks for the help!

**Comment by estruyf** [comment]
Thank @sympmarc for testing! Will release the new version tomorrow.

**Comment by sympmarc** [comment]
Thanks, @estruyf. Now that I'm finally using these controls, they are really awesome. I hope to be able to contribute more as we go along.

**Comment by vlad-nvs** [comment]
Hi all. And how about selecting SharePoint groups in this control?

Thanks Vladyslav

**Comment by estruyf** [comment]
@vlad-nvs should work as well right now. If you pass `PrincipalType.SharePointGroup` to the `principalTypes` property it should retrieve the SharePoint groups from the site.

**Comment by estruyf** [comment]
This has been released with the latest `1.11.0` version. Feel free to test it out, and you can reopen the issue if it still occurs.
